addons/point_of_sale/static/src/app/store/db.js

```diff
@@ -554,11 +554,57 @@ export class PosDB {
         }
         return list;
     }
+
+    get_product_by_template_id(tmpl_id) {
+        let list = [];
+        const products = Object.values(this.product_by_id);
+        for (const product of products) {
+            if (product.product_tmpl_id === tmpl_id) {
+                list.push(product);
+            }
+        }
+        return list;
+    }
+
+    /**
+    * Adds a product matching the given match condition (`r`) to the results array (`results`).
+    * 
+    * @param {Array} r - The result of the regular expression used to identify the product.
+    * @param {Array} results - The array where the matching product will be added.
+    * @returns {boolean} Returns true if the product was successfully added, otherwise false.
+    */
+    add_match_product(r, results){
+        if (r) {
+            let id = Number(r[1]);
+            const product = this.get_product_by_id(id);
+            if (!this.shouldAddProduct(product, results)) {
+                return false;
+            }
+            results.push(product);
+        }
+        return true;
+    }
+
     /* returns a list of products with :
      * - a category that is or is a child of category_id,
      * - a name, package or barcode containing the query (case insensitive)
      */
     search_product_in_category(category_id, query) {
+        let product_tmpl_id;
+        let filteredProducts = [];
+        try {
+            // Try to extract the product template ID from the query using a regular expression.
+            // If found, get the products filtered by the template ID and remove the ID from the query.
+            let reg = /;product\.tmpl\.id:(\d+)/gi;
+            let match = reg.exec(query);
+            if (match) {
+                product_tmpl_id = parseInt(match[1], 10);
+                filteredProducts = this.get_product_by_template_id(product_tmpl_id);
+                query = query.replace(/;product\.tmpl\.id:(\d+)/gi, '');
+            }
+        } catch (e){
+            // Handle any errors that occur during the process (optional)
```

Comment:
Except if it is intended to silence the calls, at least put something in the console
```suggestion
            console.error("Search on product template ID fails", e) // Handle any errors that occur during the process (optional)
```